lib/bolt_server/transport_app.rb

```diff
@@ -647,16 +655,23 @@ def make_winrm_target(target_hash)
     # @param project_ref [String] the project_ref to compute the inventory from
     post '/project_inventory_targets' do
       return MISSING_PROJECT_REF_RESPONSE if params['project_ref'].nil?
-      bolt_config = config_from_project(params['project_ref'])
-      if bolt_config.inventoryfile && bolt_config.project.inventory_file.to_s != bolt_config.inventoryfile
-        raise Bolt::ValidationError, "Project inventory must be defined in the " \
-          "inventory.yaml file at the root of the project directory"
-      end
-      plugins = Bolt::Plugin.setup(bolt_config, nil)
-      inventory = Bolt::Inventory.from_config(bolt_config, plugins)
-      target_list = inventory.get_targets('all').map { |targ| targ.to_h.merge({ 'transport' => targ.transport }) }
+      in_bolt_project(params['project_ref']) do |context|
+        if context[:config].inventoryfile &&
+           context[:config].project.inventory_file.to_s !=
+           context[:config].inventoryfile
+          raise Bolt::ValidationError, "Project inventory must be defined in the " \
+            "inventory.yaml file at the root of the project directory"
+        end
+        plugins = Bolt::Plugin.setup(context[:config], context[:pal], load_plugins: false)
+        begin
+          inventory = Bolt::Inventory.from_config(context[:config], plugins)
+          target_list = inventory.get_targets('all').map { |targ| targ.to_h.merge({ 'transport' => targ.transport }) }
+        rescue Bolt::Plugin::PluginError::PluginLoadingDisabled => e
+          raise Bolt::PluginNotSupported.new(e.msg, 'stub')
```

Comment:
@ekinanp This is where i would use the structured data i asked about in https://github.com/puppetlabs/bolt/pull/2394/files#r532886010

Reply:
Yup, I updated my PR to include `plugin_name` in `details`.